core/src/test/scala/unit/kafka/tools/ConsoleProducerTest.scala

```diff
@@ -123,4 +163,37 @@ class ConsoleProducerTest {
     assertEquals("console-producer",
       producerConfig.getString(ProducerConfig.CLIENT_ID_CONFIG))
   }
+
+  @Test
+  def testBatchSizeOverride(): Unit = {
+    val config = new ConsoleProducer.ProducerConfig(batchSizeOverride)
```

Comment:
Maybe rename to `testBatchSizeOverriddenByMaxPartitionMemoryBytesValue`
And the config variable name.. hmm... I think `batchSizeOverride` is OK. What do you think?